DeanerySystem/DeanerySystem.DataAccess/Utilities/DataBaseUtilities.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Linq;
+using DeanerySystem.DataAccess.Concrete;
+using DeanerySystem.DataAccess.DataFeeders;
+using DeanerySystem.DataAccess.Entities;
+
+namespace DeanerySystem.DataAccess.Utilities
+{
+    public static class DataBaseUtilities
+    {
+        public static void FeedDataBase(DeaneryDbContext context)
+        {
+            try
+            {
+                using (var uow = new UnitOfWork(context))
+                {
+                    if (uow.UniversityRepository.Get().Any())
+                    {
+                        return;                       
+                    }
+
+                    var university = new University() { Name = "ЛНУ імені Івана Франка" };
+                    uow.UniversityRepository.Insert(university);
+                    uow.Save();
+
+                    var fDataFeeder = new FacultyDataFeeder(uow);
+                    fDataFeeder.Data.ForEach(f =>
+                    {
+                        f.Id = 0;
+                        uow.FacultyRepository.Insert(f);
+                    });
+                    uow.Save();
+
+                    var sDataFeeder = new StreamDataFeeder(uow);
+                    sDataFeeder.Data.ForEach(s =>
+                    {
+                        s.Id = 0;
+                        uow.StreamRepository.Insert(s);
+                    });
+                    uow.Save();
+
+                    var dDataFeeder = new DepartmentDataFeeder(uow);
+                    dDataFeeder.Data.ForEach(d =>
+                    {
+                        d.Id = 0;
+                        uow.DepartmentRepository.Insert(d);
+                    });
+                    var semDataFeeder = new SemesterDataFeeder(uow);
+                    semDataFeeder.Data.ForEach(s =>
+                    {
+                        s.Id = 0;
+                        uow.SemesterRepository.Insert(s);
+                    });
+                    uow.Save();
+
+                    var pDataFeeder = new ProfessorDataFeeder(uow);
+                    uow.Save();
+
+                    var gDataFeeder = new GroupDataFeeder(uow);
+                    gDataFeeder.Data.ForEach(g =>
+                    {
+                        g.Id = 0;
+                        uow.GroupRepository.Insert(g);
+                    });
+                    uow.Save();
+
+                    var stDataFeeder = new StudentDataFeeder(uow);
+                    uow.Save();
```

Comment:
Shouldn't the deanery users repository be filled in with professors and students?

Reply:
It is filled in the constructor.